src/main/java/net/torocraft/flighthud/mixin/ClientPlayerEntityMixin.java

```diff
@@ -0,0 +1,26 @@
+package net.torocraft.flighthud.mixin;
+
+import net.minecraft.client.network.ClientPlayerEntity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.damage.DamageSource;
+import net.torocraft.flighthud.FlightSafetyMonitor;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(ClientPlayerEntity.class)
+public class ClientPlayerEntityMixin {
+    @Inject(method = "damage", at = @At("HEAD"))
+    public void detectWallCollision(DamageSource damageSource, float amount, CallbackInfoReturnable<Boolean> cir) {
+        LivingEntity $this = (LivingEntity) (Object) this;
+        if ($this instanceof ClientPlayerEntity
```

Comment:
Since this is a mixin to `ClientPlayerEntity`, there's no need for this `instanceof` check anymore